### PR TITLE
Issue/39 - Convert PrioritizedTask's priority field from int to enum

### LINKED ISSUE #39: Make Task Priority an Integer rather than int

### PR DESCRIPTION
**Pull request by pecjas**
Fixes issue 39.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -7,6 +7,7 @@ target
 .ear
 .ejb
 *.iml
+*.ipr
```
Comment (pecjas) [inline .gitignore:10]: Looks like the .ipr was created by intellij. Removed and added to gitignore instead.

File: analyzer-core/src/main/java/ffb/analyzer/core/queue/PrioritizedTask.java [hunk 2]
```diff
@@ -3,25 +3,43 @@
 import java.util.function.Consumer;
 
 public class PrioritizedTask<T> extends Task<T> implements Comparable<PrioritizedTask<T>> {
```
Comment (david-hu23) [inline analyzer-core/src/main/java/ffb/analyzer/core/queue/PrioritizedTask.java:10]: Can we add a Javadoc for this class?

File: analyzer-core/src/main/java/ffb/analyzer/core/queue/PrioritizedTask.java [hunk 3]
```diff
@@ -3,25 +3,43 @@
...
+    private final Priority priority;
 
     @Override
     public int compareTo(PrioritizedTask<T> otherTask) {
-        return (this.getPriority() - otherTask.getPriority());
+        return (otherTask.getPriority().getValue() - this.getPriority().getValue());
     }
 
-    public PrioritizedTask(int id, int priority, T object, Consumer<T> action) {
+    public PrioritizedTask(int id, Priority priority, T object, Consumer<T> action) {
```
Comment (david-hu23) [inline analyzer-core/src/main/java/ffb/analyzer/core/queue/PrioritizedTask.java]: Can we move the constructors up?

File: analyzer-core/src/main/java/ffb/analyzer/core/queue/PrioritizedTask.java [hunk 4]
```diff
@@ -3,25 +3,43 @@
 import java.util.function.Consumer;
 
 public class PrioritizedTask<T> extends Task<T> implements Comparable<PrioritizedTask<T>> {
-    private final Integer priority;
+    public enum Priority {
```
Comment (david-hu23) [inline analyzer-core/src/main/java/ffb/analyzer/core/queue/PrioritizedTask.java:22]: `Lowest` and `Highest`?

File: analyzer-core/src/main/java/ffb/analyzer/core/queue/PrioritizedTask.java [hunk 5]
```diff
@@ -3,25 +3,43 @@
...
+            return this.value;
+        }
+    }
+
+    private final Priority priority;
 
     @Override
     public int compareTo(PrioritizedTask<T> otherTask) {
-        return (this.getPriority() - otherTask.getPriority());
+        return (otherTask.getPriority().getValue() - this.getPriority().getValue());
```
Comment (david-hu23) [inline analyzer-core/src/main/java/ffb/analyzer/core/queue/PrioritizedTask.java]: No need for `this`. It's implied.

File: analyzer-core/src/test/java/ffb/analyzer/core/queue/PriorityQueueManagerTests.java [hunk 6]
```diff
@@ -43,13 +45,13 @@ public void performOneTask() {
     @Test
     public void performInPriorityOrder() {
```
Comment (david-hu23) [inline analyzer-core/src/test/java/ffb/analyzer/core/queue/PriorityQueueManagerTests.java]: Thoughts on making `STARTING_VALUE + 1` a local variable you compare against?

File: analyzer-core/src/test/java/ffb/analyzer/core/queue/PriorityQueueManagerTests.java [hunk 7]
```diff
@@ -24,16 +26,16 @@ public void tearDown() {
 
     @Test
     public void addToQueue() {
```
Comment (david-hu23) [inline analyzer-core/src/test/java/ffb/analyzer/core/queue/PriorityQueueManagerTests.java]: `testAddToQueue`

File: analyzer-core/src/test/java/ffb/analyzer/core/queue/PriorityQueueManagerTests.java [hunk 8]
```diff
@@ -24,16 +26,16 @@ public void tearDown() {
...
-        Assert.assertTrue(manager.addEntry(Utils.getExampleTask(0, exampleObject)));
+        Assert.assertTrue(manager.addEntry(Utils.getExampleTask(Priority.HIGHER, exampleObject)));
+        Assert.assertTrue(manager.addEntry(Utils.getExampleTask(Priority.LOW, exampleObject)));
+        Assert.assertTrue(manager.addEntry(Utils.getExampleTask(Priority.HIGHER, exampleObject)));
 
         Assert.assertEquals(EXPECTED_QUEUE_DEPTH, manager.getQueue().size());
     }
 
     @Test
     public void performOneTask() {
```
Comment (david-hu23) [inline analyzer-core/src/test/java/ffb/analyzer/core/queue/PriorityQueueManagerTests.java]: `testPerformOneTask`?

File: analyzer-core/src/test/java/ffb/analyzer/core/queue/PriorityQueueManagerTests.java (hunk 6 above)
Comment (david-hu23) [inline analyzer-core/src/test/java/ffb/analyzer/core/queue/PriorityQueueManagerTests.java]: `testPerformInPriorityOrder`?

File: .gitignore (hunk 1 above)
Comment (david-hu23) [reply .gitignore:10]: 👍

**Review by david-hu23 — CHANGES_REQUESTED** [review]

File: analyzer-core/src/main/java/ffb/analyzer/core/queue/PrioritizedTask.java (hunk 2 above)
Comment (pecjas) [reply analyzer-core/src/main/java/ffb/analyzer/core/queue/PrioritizedTask.java:10]: Updated.

File: analyzer-core/src/main/java/ffb/analyzer/core/queue/PrioritizedTask.java (hunk 3 above)
Comment (pecjas) [reply analyzer-core/src/main/java/ffb/analyzer/core/queue/PrioritizedTask.java]: Moved to top of class

File: analyzer-core/src/main/java/ffb/analyzer/core/queue/PrioritizedTask.java (hunk 4 above)
Comment (pecjas) [reply analyzer-core/src/main/java/ffb/analyzer/core/queue/PrioritizedTask.java:22]: This sort of ties back to what I mentioned on Friday about leaving room to expand. If we name them Lowest and Highest we're restricting ourselves to never have room to increase/decrease at the boundaries (unless we want Lowestest or EvenLowester - or possibly looooooooowest). I think Lower/Higher make their positions clear, while still giving us the flexibility to expand to Lowest/Highest in the future if need be with minimal impact.

File: analyzer-core/src/main/java/ffb/analyzer/core/queue/PrioritizedTask.java (hunk 5 above)
Comment (pecjas) [reply analyzer-core/src/main/java/ffb/analyzer/core/queue/PrioritizedTask.java]: Removed

File: analyzer-core/src/test/java/ffb/analyzer/core/queue/PriorityQueueManagerTests.java (hunk 6 above)
Comment (pecjas) [reply analyzer-core/src/test/java/ffb/analyzer/core/queue/PriorityQueueManagerTests.java]: Updated

File: analyzer-core/src/test/java/ffb/analyzer/core/queue/PriorityQueueManagerTests.java (hunk 8 above)
Comment (pecjas) [reply analyzer-core/src/test/java/ffb/analyzer/core/queue/PriorityQueueManagerTests.java]: Updated

File: analyzer-core/src/test/java/ffb/analyzer/core/queue/PriorityQueueManagerTests.java (hunk 7 above)
Comment (pecjas) [reply analyzer-core/src/test/java/ffb/analyzer/core/queue/PriorityQueueManagerTests.java]: Updated

File: analyzer-core/src/test/java/ffb/analyzer/core/queue/PriorityQueueManagerTests.java (hunk 6 above)
Comment (pecjas) [reply analyzer-core/src/test/java/ffb/analyzer/core/queue/PriorityQueueManagerTests.java]: I like it. It's called enough times it should have it's own label.

**Review by david-hu23 — APPROVED** [review]

**Pull request merged**